handler/middleware.go

```diff
@@ -369,6 +370,11 @@ func (m *middleware) getHeaderNames(ctx context.Context, mod wazeroapi.Module, s
 	default:
 		panic("unsupported header kind: " + strconv.Itoa(int(kind)))
 	}
+
+	for i := range names {
+		names[i] = strings.ToLower(names[i])
```

Comment:
The thought of passing a function of some sort came to mind but it seemed like a bit of an overoptimization

Reply:
this will re-allocate each time, so the optimization I guess would be a different writeNULTerminated option that converts to downcase as a part of writing. Assuming this would only be used for header keys which are ASCII that would be the cheapest thing.

So, maybe TODO on this approach as it is fun code for someone bored on a train?